application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java

```diff
@@ -78,14 +81,29 @@ public <E extends HasName, I extends EntityId> void notifyDeleteEntity(TenantId
     }
 
     public void notifyDeleteAlarm(TenantId tenantId, Alarm alarm, EntityId originatorId,
-                                  CustomerId customerId, ActionType actionType,
+                                  CustomerId customerId,
                                   List<EdgeId> relatedEdgeIds,
                                   SecurityUser user,
                                   String body, Object... additionalInfo) {
-        logEntityAction(tenantId, originatorId, alarm, customerId, actionType, user, additionalInfo);
+        logEntityAction(tenantId, originatorId, alarm, customerId, ActionType.DELETED, user, additionalInfo);
         sendAlarmDeleteNotificationMsg(tenantId, alarm, relatedEdgeIds, body);
     }
 
+    @Override
+    public void notifyDeleteRuleChain(TenantId tenantId, RuleChain ruleChain,
+                                      List<EdgeId> relatedEdgeIds, SecurityUser user) {
+        RuleChainId ruleChainId = ruleChain.getId();
+        logEntityAction(tenantId, ruleChainId, ruleChain, null, ActionType.DELETED, user, null, ruleChainId.toString());
+        if (RuleChainType.EDGE.equals(ruleChain.getType())) {
+            sendDeleteNotificationMsg(tenantId, ruleChainId, relatedEdgeIds, null);
+        }
+    }
+
+    @Override
+    public <I extends EntityId> void notifySendMsgToEdgeService(TenantId tenantId, I entityId, EdgeEventActionType edgeEventActionType) {
+            sendEntityNotificationMsg(tenantId, entityId, edgeEventActionType);
```

Comment:
incorrect tab - please replace for a single tab

Reply:
Ok